src/components/KidsForm.js

```diff
@@ -0,0 +1,173 @@
+import styled from 'styled-components';
+import AddIcon from './icons/AddIcon.js';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { nanoid } from 'nanoid';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+export default function KidsForm({ kidsData, setKidsData, setShowMessage }) {
+  const [startDate, setStartDate] = useState(new Date());
+  const [nameInput, setNameInput] = useState('');
+  const [isName, setIsName] = useState(false);
+  const navigate = useNavigate();
+
+  const options = { year: 'numeric', month: 'numeric', day: 'numeric' };
+
+  const filteredName = kidsData.filter(item => item.name.toLowerCase().trim() === nameInput.toLowerCase().trim());
+  const filteredNameAndDate = kidsData.filter(
+    item =>
+      item.birthDate === startDate.toLocaleDateString('de-DE', options) &&
+      item.name.toLowerCase().trim() === nameInput.toLowerCase().trim()
+  );
+
+  const disabledButton = nameInput.length <= 1 || startDate === null || filteredNameAndDate.length > 0 ? true : false;
+
+  function handleTextInput(e) {
+    setNameInput(e.target.value);
+  }
+
+  async function onCreateKidFormSubmit(event) {
+    event.preventDefault();
+    let id = nanoid();
+    let newDate = startDate.toLocaleDateString('de-DE', options);
+
+    setKidsData([...kidsData, { id: id, name: nameInput.trim(), birthDate: newDate }]);
+    setStartDate(new Date());
+    setNameInput('');
+    setShowMessage(true);
+    navigate('/addphrases');
+  }
+
+  function handleEmptyNameOnFocusDate() {
+    if (nameInput.length === 0) {
+      setIsName(true);
+    } else {
+      setIsName(false);
+    }
+  }
+
+  return (
+    <Wrapper>
+      <FormWrapper onSubmit={onCreateKidFormSubmit}>
+        <LabelInputText htmlFor="kids-name">Wie heisst Dein Kind?</LabelInputText>
+        <TextInput
+          type="text"
+          id="kids-name"
+          name="kids-name"
+          onChange={handleTextInput}
+          onFocus={() => setIsName(false)}
+          maxLength="20"
+          placeholder="Vorname (erforderlich)"
+          autoFocus
+          autoComplete="off"
+          value={nameInput}
+        />
+        {filteredName.length > 0 ? <ErrorMessage>Achtung! Der eingegebene Name existiert bereits!</ErrorMessage> : null}
+        {isName ? <ErrorMessage>Bitte gebe einen Namen ein!</ErrorMessage> : null}
+        {nameInput.length >= 20 ? <ErrorMessage>Die maximale Eingabe an Zeichen ist erreicht!</ErrorMessage> : null}
```

Comment:
Be careful with max and min lengths for names! Some names are longer than others 🙂

Reply:
I know ;) I´ve just did it for validation practice ;)